plasmapy/dispersion/analytical/tests/test_stix_.py

```diff
@@ -0,0 +1,108 @@
+            ({**_kwargs_single_valued, "w": [-1, 2] * u.rad / u.s}, ValueError),
+            ({**_kwargs_single_valued, "w": np.ones((2, 2)) * u.rad / u.s}, ValueError),
+            ({**_kwargs_single_valued, "w": 5 * u.s}, u.UnitTypeError),
+            (
+                {**_kwargs_single_valued, "ions": {"not": "a particle"}},
+                InvalidParticleError,
+            ),
+            ({**_kwargs_single_valued, "n_i": "wrong type"}, TypeError),
+            ({**_kwargs_single_valued, "n_i": 6 * u.m / u.s}, u.UnitTypeError),
+            ({**_kwargs_single_valued, "theta": 5 * u.eV}, u.UnitTypeError),
+            (
+                {**_kwargs_single_valued, "theta": np.ones((2, 2)) * u.rad},
+                TypeError,
+            ),
+            ({**_kwargs_single_valued, "ions": Particle("e-")}, ValueError),
+            ({**_kwargs_single_valued, "n_i": [4, 2, 3] * u.m ** -3}, ValueError),
+            (
+                {**_kwargs_single_valued, "n_i": np.ones((2, 2)) * u.m ** -3},
+                ValueError,
+            ),
+        ],
+    )
+    def test_raises(self, kwargs, _error):
+        with pytest.raises(_error):
+            stix(**kwargs)
+
+    @pytest.mark.parametrize(
+        "kwargs, expected",
+        [
+            ({**_kwargs_single_valued, "w": 2 * u.rad / u.s}, {"shape": (4,)}),
+            (
+                {**_kwargs_single_valued, "w": [10] * u.rad / u.s},
+                {"shape": (4,)},
+            ),
+            (
+                {**_kwargs_single_valued, "w": [10, 20, 30] * u.rad / u.s},
+                {"shape": (3, 4)},
+            ),
+            ({**_kwargs_single_valued, "ions": ["He+", "H+"]}, {"shape": (4,)}),
+            (
+                {
+                    **_kwargs_single_valued,
+                    "ions": ["He+"],
+                    "n_i": [1] * u.m ** -3,
+                },
+                {"shape": (4,)},
+            ),
+            ({**_kwargs_single_valued, "ions": ["He+", "H+"]}, {"shape": (4,)}),
+            (
+                {
+                    **_kwargs_single_valued,
+                    "ions": ["He+", "H+"],
+                    "n_i": [1, 2] * u.m ** -3,
+                },
+                {"shape": (4,)},
+            ),
+            ({**_kwargs_single_valued, "theta": [10, 20, 30]}, {"shape": (3, 4)}),
+            (
+                {
+                    **_kwargs_single_valued,
+                    "w": [10, 20],
+                    "theta": 10 * u.rad,
+                },
+                {"shape": (2, 4)},
+            ),
+        ],
+    )
+    def test_return_structure(self, kwargs, expected):
+        k = stix(**kwargs)
+
+        assert isinstance(k, u.Quantity)
+        assert np.shape(k) == expected["shape"]
+        assert k.unit == u.rad / u.m
+
+    @pytest.mark.parametrize(
+        "kwargs, expected",
+        [({**_kwargs_single_valued, "w": 1 * u.rad / u.s}, u.Quantity)],
+    )
+    def test_vals(self, kwargs, expected):
```

Comment:
For testing values I think we should do something similar to what we did for Hollweg (PR #1189).  That would be to use  [WebPlotDigitizer](https://github.com/ankitrohatgi/WebPlotDigitizer) on figures 1-1 to 1-5 in Stix to get values and then test on those.

https://books.google.com/books?id=OsOWJ8iHpmMC&lpg=PA14&pg=PA14#v=onepage&q&f=false